src/solvers/cma_es.jl

```diff
@@ -0,0 +1,851 @@
+#
+# State
+#
+@doc raw"""
+    CMAESState{P,T} <: AbstractManoptSolverState
+
+State of covariance matrix adaptation evolution strategy.
+
+# Fields
+
+* `p`                           the best point found so far
+* `p_obj`                       objective value at `p`
+* `μ`                           parent number
+* `λ`                           population size
+* `μ_eff`                       variance effective selection mass for the mean
+* `c_1`                         learning rate for the rank-one update
+* `c_c`                         decay rate for cumulation path for the rank-one update
+* `c_μ`                         learning rate for the rank-μ update
+* `c_σ`                         decay rate for the cumulation path for the step-size control
+* `c_m`                         learning rate for the mean
+* `d_σ`                         damping parameter for step-size update
+* `stop`                        stopping criteria
+* `population`                  population of the current generation
+* `ys_c`                        coordinates of random vectors for the current generation
+* `covariance_matrix`           coordinates of the covariance matrix
+* `covariance_matrix_eigen`     eigendecomposition of covariance_matrix
+* `covariance_matrix_cond`      condition number of covariance_matrix, updated after eigendecomposition
+* `best_fitness_current_gen`    best fitness value of individuals in the current generation
+* `median_fitness_current_gen`  median fitness value of individuals in the current generation
+* `worst_fitness_current_gen`   worst fitness value of individuals in the current generation
+* `p_m`                         point around which we search for new candidates
+* `σ`                           step size
+* `p_σ`                         coordinates of a vector in ``T_{p_m} \mathcal M``
+* `p_c`                         coordinates of a vector in ``T_{p_m} \mathcal M``
+* `deviations`                  standard deviations of coordinate RNG
+* `buffer`                      buffer for random number generation and `wmean_y_c` of length `n_coords`
+* `e_mv_norm`                   expected value of norm of the `n_coords`-variable standard normal distribution
+* `recombination_weights`       recombination weights used for updating covariance matrix
+* `retraction_method`
+* `vector_transport_method`
+* `basis`
+* `rng`
```

Comment:
These 4 lines still need docs, but you can nearly steal that from other solvers.